Sming/third-party/.patches/esp-open-lwip.patch

```diff
@@ -343,3 +385,165 @@ index af6e360..71b2178 100644
+@@ -96,16 +96,17 @@ struct mdns_info {
+ 	char *txt_data[10];
+ };
+ #endif
+-//void 		   mdns_enable(void);
+-//void           mdns_disable(void);
+-//void           mdns_init(struct mdns_info *info);
+-//void           mdns_close(void);
+-//char* 		   mdns_get_hostname(void);
+-//void           mdns_set_hostname(char *name);
+-//void           mdns_set_servername(const char *name);
+-//char*          mdns_get_servername(void);
+-//void           mdns_server_unregister(void);
+-//void           mdns_server_register(void) ;
++
++void 		   mdns_enable(void);
++void           mdns_disable(void);
++void           mdns_init(struct mdns_info *info);
++void           mdns_close(void);
++char* 		   mdns_get_hostname(void);
++void           mdns_set_hostname(char *name);
++void           mdns_set_servername(const char *name);
++char*          mdns_get_servername(void);
++void           mdns_server_unregister(void);
++void           mdns_server_register(void) ;
+ //void           mdns_tmr(void);
+ //void           Delay(unsigned long ulSeconds);
+ 
+diff --git a/lwip/app/espconn_tcp.c b/lwip/app/espconn_tcp.c
+index 52cd0b0..4e59429 100644
+--- a/lwip/app/espconn_tcp.c
++++ b/lwip/app/espconn_tcp.c
+@@ -557,7 +557,7 @@ espconn_recv_hold(struct espconn *pespconn)
+     value = espconn_find_connection(pespconn, &pnode);
+ 	if(value != true)
+ 	{
+-		os_printf("RecvHold, By pespconn,find conn_msg fail\n");
++		os_printf(_F("RecvHold, By pespconn,find conn_msg fail\n"));
+ 		return ESPCONN_ARG;
+ 	}
+ 
+@@ -582,7 +582,7 @@ espconn_recv_unhold(struct espconn *pespconn)
+     value = espconn_find_connection(pespconn, &pnode);
+ 	if(value != true)
+ 	{
+-		os_printf("RecvHold, By pespconn,find conn_msg fail\n");
++		os_printf(_F("RecvHold, By pespconn,find conn_msg fail\n"));
+ 		return ESPCONN_ARG;
+ 	}
+ 
+@@ -862,7 +862,7 @@ espconn_client_connect(void *arg, struct tcp_pcb *tpcb, err_t err)
+ 			espconn_keepalive_enable(tpcb);
+ 
+     } else{
+-    	os_printf("err in host connected (%s)\n",lwip_strerr(err));
++    	os_printf(_F("err in host connected (%s)\n"),lwip_strerr(err));
+     }
+     return err;
+ }
+@@ -1354,7 +1354,7 @@ sint8 ICACHE_FLASH_ATTR espconn_tcp_delete(struct espconn *pdeletecon)
+ 				/*remove the node from the client's active connection list*/
+ 				espconn_list_delete(&pserver_list, pdelete_msg);
+ 				pcb = pdelete_msg->preverse;
+-				os_printf("espconn_tcp_delete %d, %d\n",pcb->state, pcb->local_port);
++				os_printf(_F("espconn_tcp_delete %d, %d\n"),pcb->state, pcb->local_port);
+ 				espconn_kill_pcb(pcb->local_port);
+ 				err = tcp_close(pcb);
+ 				os_free(pdelete_msg);
+
+diff --git a/lwip/app/espconn_udp.c b/lwip/app/espconn_udp.c
+index 77ef471..7d6705e 100644
+--- a/lwip/app/espconn_udp.c
++++ b/lwip/app/espconn_udp.c
+@@ -19,9 +19,18 @@
+ #include "lwip/mem.h"
+ #include "lwip/tcp_impl.h"
+ #include "lwip/udp.h"
++#include "lwip/igmp.h"
+ 
+ #include "lwip/app/espconn_udp.h"
+ 
++
++/*
++ * wifi_get_ip_info()
```

Comment:
Can you put a bit more information in the comment? Or just remove it if it does not contain useful information.